pkg/downloader/manager.go

```diff
@@ -896,3 +906,54 @@ func key(name string) (string, error) {
 	}
 	return hex.EncodeToString(hash.Sum(nil)), nil
 }
+
+// dependenciesToResolve takes the requested (req) dependencies from Chart.yaml and
+// overrides the version with the matching dependency from Chart.lock, except for those
+// dependencies which have been selected for update by the user.
+// The intention is to allow a user to update only selected dependencies in Chart.yaml
+// If no selected deps are specified, returns the original request to update all dependencies.
+// Returns an error if lock is nil or if some any unselected dependencies are missing
+// from Chart.lock.
+func dependenciesToResolve(req []*chart.Dependency, lock *chart.Lock, selected []string) ([]*chart.Dependency, error) {
+	if len(selected) == 0 {
+		return req, nil
+	}
+	if lock == nil {
+		return nil, fmt.Errorf("file Chart.lock not found, the '-d/--dependency' requires an existing Chart.lock file")
+	}
+	depsToUpdate := []*chart.Dependency{}
+	for _, dep := range req {
+		if stringSliceContains(selected, dep.Name) {
```

Comment:
A chart can have multiple dependencies of the same name, but with different [aliases](https://github.com/helm/helm/blob/42ef0fe2ff18935ee9600de700b1ac0a6f54850e/pkg/chart/dependency.go#L49).

Shouldn't this logic should account for aliases? I think in that if an alias is present, that should be matched, rather than the name.